src/org/opendatakit/briefcase/reused/UncheckedFiles.java

```diff
@@ -0,0 +1,271 @@
+    }
+  }
+
+  public static Path copy(Path source, Path target, CopyOption... options) {
+    try {
+      return Files.copy(source, target, options);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  public static Path createTempDirectory(Path dir, String prefix, FileAttribute<?>... attrs) {
+    try {
+      return Files.createTempDirectory(dir, prefix, attrs);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  public static Path createTempDirectory(String prefix, FileAttribute<?>... attrs) {
+    try {
+      return Files.createTempDirectory(prefix, attrs);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  public static byte[] readAllBytes(Path path) {
+    try {
+      return Files.readAllBytes(path);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  public static String getMd5Hash(Path file) {
+    try {
+      // CTS (6/15/2010) : stream file through digest instead of handing
+      // it the
+      // byte[]
+      MessageDigest md = MessageDigest.getInstance("MD5");
+      int chunkSize = 256;
+
+      byte[] chunk = new byte[chunkSize];
+
+      // Get the size of the file
+      long lLength = Files.size(file);
+
+      if (lLength > Integer.MAX_VALUE) {
+        log.error("File is too large");
+        return null;
+      }
+
+      int length = (int) lLength;
+
+      InputStream is = Files.newInputStream(file);
+
+      int l = 0;
+      for (l = 0; l + chunkSize < length; l += chunkSize) {
+        is.read(chunk, 0, chunkSize);
+        md.update(chunk, 0, chunkSize);
+      }
+
+      int remaining = length - l;
+      if (remaining > 0) {
+        is.read(chunk, 0, remaining);
+        md.update(chunk, 0, remaining);
+      }
+      byte[] messageDigest = md.digest();
+
+      BigInteger number = new BigInteger(1, messageDigest);
+      String md5 = number.toString(16);
+      while (md5.length() < 32)
+        md5 = "0" + md5;
+      is.close();
+      return md5;
+
+    } catch (NoSuchAlgorithmException e) {
+      log.error("MD5 calculation failed", e);
+      return null;
+
+    } catch (FileNotFoundException e) {
+      log.error("No File", e);
+      return null;
+    } catch (IOException e) {
+      log.error("Problem reading from file", e);
+      return null;
+    }
+
+  }
+
+  public static String stripFileExtension(String fileName) {
+    return fileName.contains(".")
+        ? fileName.substring(0, fileName.lastIndexOf("."))
+        : fileName;
+  }
+
+  public static long checksumOf(Path file) {
+    try {
+      return FileUtils.checksumCRC32(file.toFile());
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  public static boolean isInstanceDir(Path dir) {
+    return Files.isDirectory(dir)
+        && dir.getFileName().toString().matches("^uuid[\\w]{8}-[\\w]{4}-[\\w]{4}-[\\w]{4}-[\\w]{12}$")
```

Comment:
This is a little scary. Will it break if the naming convention changes? Should this test reside closer to where the convention is implemented?
It would be helpful to at least add a comment with an example of an instance directory so readers can match it up with the regex if they want to.

Reply:
> This is a little scary. Will it break if the naming convention changes? Should this test reside closer to where the convention is implemented?

That's a very good point! I'll add a comment and we can open an issue to fix this design flaw after we get this merged. This affects the pulling process where these directories are created

Reply:
That’s a good comment.